src/seedu/addressbook/commands/Command.java

```diff
@@ -63,6 +63,10 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    public boolean isMutating() {
+        return true;
```

Comment:
Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.